haystack/preview/components/fetchers/link_content.py

```diff
@@ -0,0 +1,225 @@
+import inspect
+import io
+import logging
+from collections import defaultdict
+from datetime import datetime
+from http import HTTPStatus
+from typing import Optional, Dict, List, Callable, Any, IO
+from urllib.parse import urlparse
+
+import requests
+from requests import Response
+from requests.exceptions import InvalidURL, HTTPError
+from tenacity import retry, stop_after_attempt, wait_exponential, retry_if_exception_type, RetryCallState
+from haystack.preview import component, default_from_dict, default_to_dict
+
+from haystack import __version__
+from haystack.preview import Document
+
+logger = logging.getLogger(__name__)
+
+
+def text_content_handler(response: Response) -> Optional[str]:
+    """
+    :param response: Response object from the request.
+    :return: The extracted text.
+    """
+    return response.text
+
+
+def binary_content_handler(response: Response) -> IO[bytes]:
+    """
+    :param response: Response object from the request.
+    :return: The extracted binary file-like object.
+    """
+    return io.BytesIO(response.content)
+
+
+@component
+class LinkContentFetcher:
+    """
+    LinkContentFetcher fetches content from a URL link
+    """
+
+    outgoing_edges = 1
+
+    _USER_AGENT = f"haystack/LinkContentRetriever/{__version__}"
+
+    _REQUEST_HEADERS = {
+        "accept": "*/*",
+        "User-Agent": _USER_AGENT,
+        "Accept-Language": "en-US,en;q=0.9,it;q=0.8,es;q=0.7",
+        "referer": "https://www.google.com/",
+    }
+
+    def __init__(
+        self,
+        raise_on_failure: Optional[bool] = False,
```

Comment:
I would prefer this to default to True, users can then set it to False when they're more confident their system works. wdyt?